src/nxt_http_route.c

```diff
@@ -2134,6 +2134,10 @@ nxt_http_route_pattern(nxt_http_request_t *r, nxt_http_route_pattern_t *pattern,
         return 0;
     }
 
+    if (nxt_slow_path(length == 0)) {
+        return 1;
+    }
+
```

Comment:
Again, if this is because _start_ can be NULL, then check for that.

The reason I'm saying this is because it's not that if _length == 0_ is the problem, it's that if the pointer you're applying it to is NULL.

This is all assuming these things can be NULL, are we sure of that?

Reply:
However, in _this_ instance, I'd be willing to accept the above check given the commit message.

But I still think the NULL checks are more appropriate where I've mentioned elsewhere.

Reply:
Hmm, looking at the code again, I think I'd check if start is NULL.